### PR TITLE
chore: Misc. improvements to e2e-kind.sh

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What type of PR is this?**
Chore

**What this PR does/ why we need it**:
- Prevent the containerized e2e test from polluting host directory
- Wait for deployments to be ready
- Fail on errors in bash pipelines in subshells
- Respect git remote name in the e2e-kind.sh script (it is already respected in the Makefile)

**Which issue(s) this PR fixes**:
no issue

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Checklist**

* [x] The commit message explains the changes and why are needed.

### CODE REVIEW

File: test/e2e-kind.sh [hunk 1]
```diff
@@ -53,7 +58,7 @@ create_kind_cluster() {
 
     # if we are running inside of a kubernetes cluster with /kubepods cgroup
     # being used for pods --> add kubeadmConfigPatches
-    if cat /proc/1/cgroup 2>/dev/null | tail -1 | grep -q /kubepods; then
+    if tail -1 /proc/1/cgroup 2>/dev/null | grep -q /kubepods; then
```
Comment (makkes) [inline test/e2e-kind.sh:61]: yay, one [less useless use of cat](https://en.wikipedia.org/wiki/Cat_(Unix)#Useless_use_of_cat)

**Comment by makkes** [comment]
@dlipovetsky for some reason, Dispatch isn't starting thee check runs or not properly reporting their results...

File: test/e2e-kind.sh [hunk 2]
```diff
@@ -22,13 +23,17 @@ run_ct_container() {
...
-        --volume "$(pwd):/workdir" \
         "${teamcity_volume[@]}" \
-        --workdir /workdir \
+        --workdir /charts \
         "quay.io/helmpack/chart-testing:${CT_VERSION}" \
         cat
     echo
+
+    docker cp "$(pwd)/test/ct-e2e.yaml" "ct:/etc/ct/ct.yaml"
+    docker cp "$(pwd)" "ct:/charts"
```
Comment (makkes) [inline test/e2e-kind.sh:33]:
```suggestion
    docker cp "$(pwd)/" "ct:/charts"
```
This should fix it.

Comment (shaneutt) [reply test/e2e-kind.sh:33]: let's give that a shot thanks max :+1:

**Comment by makkes** [comment]
```
+ docker exec --interactive ct ls -l /charts
total 0
drwxr-xr-x    8 root     root           231 Jul 30 14:58 src-git
```
dang, didn't change anything. 🤔

**Comment by dlipovetsky** [comment]
Something is wrong with the way the repo is copied inside the container. Strangely, I haven't been able to reproduce it locally. I'll try again today. I think I may just remove the commit for that.

**Comment by dlipovetsky** [comment]
I think I know what's going on. As a result of `--workdir /charts`, the `/charts` directory is created, so I need to copy `charts` into the container root directory. Will fix.

**Comment by sebbrandt87** [comment]
:Love:

File: test/e2e-kind.sh (hunk 2 above)
Comment (dlipovetsky) [reply test/e2e-kind.sh]: Thanks for trying @makkes :slightly_smiling_face:  Turns out I needed to copy everything _in_ the current working directory, i.e., `$pwd)/.`

Can you approve?

File: test/e2e-kind.sh [hunk 3]
```diff
@@ -22,13 +23,17 @@ run_ct_container() {
...
-        --workdir /workdir \
+        --workdir /charts \
         "quay.io/helmpack/chart-testing:${CT_VERSION}" \
         cat
     echo
+
+    docker cp "$(pwd)/test/ct-e2e.yaml" "ct:/etc/ct/ct.yaml"
+    docker cp "$(pwd)/." "ct:/charts"
+    docker_exec chown -R root:root /charts
+    # TODO(dlipovetsky) Debugging
```
Comment (makkes) [inline test/e2e-kind.sh:35]: Do you want to remove that before merging?

Comment (dlipovetsky) [reply test/e2e-kind.sh:35]: :facepalm: Thanks, will fix!

**Pull request merged**